core/templates/filters/translate.pipe.spec.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the translate pipe.
+ */
+
+import { async, TestBed } from '@angular/core/testing';
+import { EventEmitter, ChangeDetectorRef } from '@angular/core';
+
+import { TranslateService } from 'services/translate.service';
+import { TranslatePipe } from './translate.pipe';
+import { UtilsService } from 'services/utils.service';
+
+
+class MockChangeDetectorRef {
+  markForCheck() {
+    return true;
+  }
+}
+
+class MockTranslateService {
+    langChange = new EventEmitter();
```

Comment:
This is not named right. "EventEmitter" should be in the name, otherwise this looks like some kind of domain object.

Reply:
Changed!